+++
title = "lftp [FEAT negotiation...]"
postid = 2654
date = 2017-10-08T17:09:11+08:00
isCJKLanguage = true
toc = false
type = "post"
slug = "lftp-feat-negotiation"
aliases = [ "/post/2654.html",]
category = [ "technology",]
tag = [ "server", "ftp",]
+++


[lftp][lftp] 是我一直在使用的 ftp 客户端。今天将其更新到 4.8.2 之后，出现了下面的错误： <!--more-->

```
<--- 220---------- Welcome to Pure-FTPd [privsep] [TLS] ----------
<--- 220-You are user number 3 of 50 allowed.
<--- 220-Local time is now 16:59. Server port: 21.
<--- 220-This is a private system - No anonymous login
<--- 220-IPv6 connections are also welcome on this server.
<--- 220 You will be disconnected after 15 minutes of inactivity.
[FEAT negotiation...]
```

提示符停止 `[FEAT negotiation...]` 就不再继续了。

直接使用 `ftp` 命令是可以正常登录服务器的，这说明服务器本身没有问题。

FEAT 是一个 FTP 扩展，定义在 [RFC2389][rfc2389] 中。它的作用是获取服务器支持的特性列表。

从日志中可以看到，lftp 在鉴权之前就调用了 FEAT 命令，这应该是服务器拒绝登录的原因。

**解决方案：**

编辑配置文件 `~/.config/lftp/rc` ，加入下面的配置：

```
set use-feat no
```

{% label '全文完' %}

[lftp]: https://lftp.yar.ru/
[rfc2389]: https://tools.ietf.org/html/rfc2389